### PR TITLE
Links Security Status to Sechuds

### PR DESCRIPTION
**Pull request by thedarkcrusader**
## About The Pull Request
This PR  links  the security record. field from crew records to sechuds allowing them to be seen by those wearing sechuds

## Why It's Good For The Game
Allows for much Better   in field keeping up to date with incidents. and problems without having to run to a computer every 
image

single time just to see a person's history.  

## Testing
spawned in as operative.
 edited secrecords from console
viewed myself with sechuds
Records appeared as normal 

## Changelog
:cl:
tweak: sechuds now draw their information from the crew record program. 
/:cl:

### CODE REVIEW

**Comment by thevandie** [comment]
Failed integration test, L bozo

**Comment by thedarkcrusader** [comment]
> Contributor

Intergration failed??  why don't you go inter into some bitches homes.

**Review by Firefox13 — COMMENTED** [review]
> Records appeared as normal

How does it work with old chat (believe it or not, it needs to work with old chat), how it works if record contain Details (stuff you add in character setup)?

**Comment by thedarkcrusader** [comment]
> > Records appeared as normal
> 
> How does it work with old chat (believe it or not, it needs to work with old chat), how it works if record contain Details (stuff you add in character setup)?
Anything added to security status in setup will appear let me check it in old chat rq tho should still be fine

**Comment by thedarkcrusader** [comment]
> > Records appeared as normal
> 
> How does it work with old chat (believe it or not, it needs to work with old chat), how it works if record contain Details (stuff you add in character setup)?


still works with Old chat but the some of the HTML  code like boxes don't show up properly. otherwise works as normal

**Comment by Firefox13** [comment]
Screenshots please?

**Comment by thedarkcrusader** [comment]
> Screenshots please?

image

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -367,19 +367,16 @@ var/list/rank_prefix = list(\
...
+			if(E)
+				// Check if user has security HUD again
+				if(hasHUD(usr,"security"))
+					to_chat(usr, "<b>Name:</b> [E.get_name()]")
+					to_chat(usr, "<b>Criminal Status:</b> [E.get_criminalStatus()]")
+					to_chat(usr, "<b>Details:</b> [E.get_secRecord()]")
+					read = 1
 
 			if(!read)
 				to_chat(usr, "\red Unable to locate a data core entry for this person.")
```
Comment (SirRichardFrancis) [inline code/modules/mob/living/carbon/human/human.dm:382]:
```suggestion
			var/datum/computer_file/report/crew_record/crew_record = get_crewmember_record(perpname)
			if(crew_record && hasHUD(usr, "security"))
				to_chat(usr, "<b>Name:</b> [crew_record.get_name()]")
				to_chat(usr, "<b>Criminal Status:</b> [crew_record.get_criminalStatus()]")
				to_chat(usr, "<b>Details:</b> [crew_record.get_secRecord()]")
			else
				to_chat(usr, span_red("Unable to locate a data core entry for this person."))
```
`read` variable is redundant, delete it from this scope. It's too far away to be included in a code suggestion.

File: code/modules/mob/living/carbon/human/human.dm [hunk 2]
```diff
@@ -1348,7 +1345,7 @@ var/list/rank_prefix = list(\
 	reset_view(A)
 
 /mob/living/carbon/human/proc/resuscitate()
-	
+
```
Comment (SirRichardFrancis) [inline code/modules/mob/living/carbon/human/human.dm:1348]:
```suggestion
```

**Review by SirRichardFrancis — CHANGES_REQUESTED** [review]
image
This is no bueno, spacings are so massive that half the chat is blank.
Even if some information is meant to be there, text should be compressed for cases when there isn't any.

**Comment by thedarkcrusader** [comment]
> image This is no bueno, spacings are so massive that half the chat is blank. Even if some information is meant to be there, text should be compressed for cases when there isn't any.

Those  spaces are from my HTML layout

**Comment by SirRichardFrancis** [comment]
> Those spaces are from my HTML layout

So? Make them disappear.

**Comment by thedarkcrusader** [comment]
> > Those spaces are from my HTML layout
> 
> So? Make them disappear.

Allright it's gone

**Comment by thedarkcrusader** [comment]
Poo

> image This is no bueno, spacings are so massive that half the chat is blank. Even if some information is meant to be there, text should be compressed for cases when there isn't any.

On a more serious note would you by. Chance know how one could do that compression??

**Comment by SirRichardFrancis** [comment]
> On a more serious note would you by. Chance know how one could do that compression??

As a matter of fact, I can't find the code responsible for this, since text from your screenshots does not appear to be in the build, and neither is added by this PR.
You're sure that no changes remain uncommitted?

**Comment by thedarkcrusader** [comment]
> > On a more serious note would you by. Chance know how one could do that compression??
> 
> As a matter of fact, I can't find the code responsible for this, since text from your screenshots does not appear to be in the build, and neither is added by this PR. You're sure that no changes remain uncommitted?

Im positive which text are you referring to ?

**Comment by SirRichardFrancis** [comment]
> Im positive which text are you referring to ?

Anything from the screenshot you put in PR description.
image

For example let's take last table header.
image

Or some of the blue text.
image
 
Other word perhaps?
image

There could be all sorts of weird formatting in strings, which could make them borderline unsearchable, but individual words always fair game. Yet there isn't any in the code.
This table and text around it doesn't seem to exist anywhere.
So I have absolutely no clue where this table is coming from, as such can't help with formatting it properly.

**Comment by SirRichardFrancis** [comment]
If you made that table in your character's records, then this is nonsensical feature presentation.

**Comment by thedarkcrusader** [comment]
> If you made that table in your character's records, then this is nonsensical feature presentation.

I mean  how would you expect me to show it off this is quite literally  one of the things people are able to do now

**Comment by thedarkcrusader** [comment]
,
image


> If you made that table in your character's records, then this is nonsensical feature presentation.

**Comment by SirRichardFrancis** [comment]
> I mean how would you expect me to show it off this is quite literally one of the things people are able to do now

~~Using something realistic, like what an actual player, that seen the sun at least once in past three years, could write?~~
PR description made me think that every sechud examine will have that cursed HTML table.
Since that's apparently just a skill issue on the player part, I have no objections.

**Comment by MLGTASTICa** [comment]
Note for maintainers , someone should really check if sec records are actually sanitized in the field that this PR outputs , else someone could inject JS.

**Comment by SirRichardFrancis** [comment]
> why is this conflicting?

Because you and someone else (who got their PR merged first) have edited the same file.
Just pull upstream/master on your local branch, solve the conflict in VSC, save file, git add, git commit, and git push.
5 minutes adventure.
image

**Comment by thedarkcrusader** [comment]
> > why is this conflicting?
> 
> Because you and someone else (who got their PR merged first) have edited the same file. Just pull upstream/master on your local branch, solve the conflict in VSC, save file, git add, git commit, and git push. 5 minutes adventure. image

Ahh jeez rick i don't want to go on a 5 minute adventure

**Comment by Humonitarian** [comment]
Merge conflicts still present.

**Comment by thedarkcrusader** [comment]
> Merge conflicts still present.

im prob gonna kill this i think there is a better way to do this

**Comment by thedarkcrusader** [comment]
Bye bye conflicts

**Pull request closed without merging**